.github/workflows/test.yml

```
@@ -0,0 +1,90 @@
+name: build-and-test
+on: [pull_request, push]
```

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```
I think this is the strategy to avoid the double CI for development branches on upstream